File: CS/SnapAndSpreadsheetDocumentServerExample/XDataSourceHelper.cs
using DevExpress.Spreadsheet;

namespace SnapAndSpreadsheetDocumentServerExample {
    #region #XDataSourceHelper
    public static class XDataSourceHelper {
        public static object GetData() {
            Workbook wb = new Workbook();
            wb.LoadDocument("Employees.xlsx");
            RangeDataSourceOptions options = new RangeDataSourceOptions();
            options.UseFirstRowAsHeader = true;
            options.CellValueConverter = new XPictureProvider(wb.Worksheets[0]);
            return wb.Worksheets[0].Tables[0].GetDataSource(options);
        }
    }
    #endregion #XDataSourceHelper
}

File: VB/SnapAndSpreadsheetDocumentServerExample/XDataSourceHelper.vb
Imports DevExpress.Spreadsheet

Namespace SnapAndSpreadsheetDocumentServerExample
    #Region "#XDataSourceHelper"
    Public NotInheritable Class XDataSourceHelper

        Private Sub New()
        End Sub

        Public Shared Function GetData() As Object
            Dim wb As New Workbook()
            wb.LoadDocument("Employees.xlsx")
            Dim options As New RangeDataSourceOptions()
            options.UseFirstRowAsHeader = True
            options.CellValueConverter = New XPictureProvider(wb.Worksheets(0))
            Return wb.Worksheets(0).Tables(0).GetDataSource(options)
        End Function
    End Class
    #End Region ' #XDataSourceHelper
End Namespace
